### PR TITLE
Loading CSAF document from String

### PR DESCRIPTION
**Pull request by milux**
- **Add `fromJson` method and related tests for JSON parsing**
- **Refactor Result to ResultCompat and relocate RetrievedDocument.**

### CODE REVIEW

**Comment by milux** [comment]
Implements #118 with `String` input.

**Comment by oxisto** [comment]
cc @lawid. Is `String` ok?

**Comment by lawid** [comment]
I guess its fine but i'm worried about large documents, although i suppose the entire document will be parsed anyway right?

**Review by lawid — APPROVED** [review]

File: csaf-retrieval/src/jvmMain/kotlin/io/github/csaf/sbom/retrieval/RetrievedDocument.kt [hunk 1]
```diff
@@ -61,5 +62,13 @@ data class RetrievedDocument(val json: Csaf, @Suppress("unused") val sourceUrl:
                     throw Exception("Failed to load CSAF document from $documentUrl", e)
                 }
         }
+
+        fun fromJson(json: String): ResultCompat<RetrievedDocument> {
+            return try {
+                ResultCompat.success(RetrievedDocument(Json.decodeFromString<Csaf>(json)))
```
Comment (oxisto) [inline csaf-retrieval/src/jvmMain/kotlin/io/github/csaf/sbom/retrieval/RetrievedDocument.kt:68]: Any particular reason not to expose `Json.decodeFromStream`?

Comment (milux) [reply csaf-retrieval/src/jvmMain/kotlin/io/github/csaf/sbom/retrieval/RetrievedDocument.kt:77]: Stream is JVM-specific, but I included both now.

File: csaf-retrieval/src/jvmMain/kotlin/io/github/csaf/sbom/retrieval/RetrievedDocument.kt [hunk 2]
```diff
@@ -61,5 +62,13 @@ data class RetrievedDocument(val json: Csaf, @Suppress("unused") val sourceUrl:
                     throw Exception("Failed to load CSAF document from $documentUrl", e)
                 }
         }
+
+        fun fromJson(json: String): ResultCompat<RetrievedDocument> {
```
[the remark this answers is not in the document]
Comment (milux) [reply csaf-retrieval/src/jvmMain/kotlin/io/github/csaf/sbom/retrieval/RetrievedDocument.kt:75]: KDoc. :wink: Right though, done. :+1:

File: csaf-retrieval/src/jvmMain/kotlin/io/github/csaf/sbom/retrieval/RetrievedDocument.kt (hunk 1 above)
Comment (milux) [reply csaf-retrieval/src/jvmMain/kotlin/io/github/csaf/sbom/retrieval/RetrievedDocument.kt:77]: It's experimental, BTW. Living on the edge. :laughing:

**Review by oxisto — APPROVED** [review]

**Pull request merged**